### PR TITLE
Use React Suspense when rendering async modals

### PR DESCRIPTION
**Pull request by t3chguy**

### CODE REVIEW

**Review by richvdh — COMMENTED** [review]
What's React Suspense? Why is it good, and what are the practical implications of it?

**Comment by t3chguy** [comment]
> What's React Suspense?

https://react.dev/reference/react/Suspense

> Why is it good

Enables Concurrent Mode https://react.dev/blog/2022/03/29/react-v18#what-is-concurrent-react

and what are the practical implications of it?

1. Aligning with the React way to do things
2. Fulfilling a customer contract
3. Speed

**Comment by richvdh** [comment]
I'm going to wait for review from a member of the web team before looking at this. If the web team is happy, I'll check the crypto bits.

**Review by florianduros — APPROVED** [review]

File: src/Modal.tsx
Comment (richvdh) [inline src/Modal.tsx:175]: are A and P actually used here?

**Review by richvdh — APPROVED** [review]
lgtm

File: src/Modal.tsx
Comment (t3chguy) [reply src/Modal.tsx:175]: Whoops this comment is stale from when I tried to make the types even more rigid and failed

```suggestion
```

**Pull request merged**
